test/cpp/tensorexpr/test_kernel.cpp

```diff
@@ -853,5 +842,88 @@ void testKernelSoftmax4D() {
   }
 }
 
+void testKernelInlineProducerIntoReduction() {
```

Comment:
I understand adding a test in kernel because of the change there, but all the reduction tests exist in test_reduction. are you sure this is covered there also?

Reply:
Yes. Here are the 3 tests that cover these cases in `test_reductions.cpp`. 

Positive tests: [`testReduceInlineConsumer`](https://github.com/pytorch/pytorch/blob/master/test/cpp/tensorexpr/test_reductions.cpp#L1344), [`testReduceInlineReducerInternal`](https://github.com/pytorch/pytorch/blob/master/test/cpp/tensorexpr/test_reductions.cpp#L1398)

Negative test: [`testReduceInlineReduction`](https://github.com/pytorch/pytorch/blob/master/test/cpp/tensorexpr/test_reductions.cpp#L1311)

Reply:
Oh write, yeah, I wrote those tests I should remember :)